core/commands/config.go

```diff
@@ -75,21 +75,14 @@ Set the value of the 'datastore.path' key:
 		var output *ConfigField
 		if len(args) == 2 {
 			value := args[1]
-
-			if parseJson, _, _ := req.Option("json").Bool(); parseJson {
-				var jsonVal interface{}
-				if err := json.Unmarshal([]byte(value), &jsonVal); err != nil {
-					err = fmt.Errorf("failed to unmarshal json. %s", err)
-					res.SetError(err, cmds.ErrNormal)
-					return
-				}
-
-				output, err = setConfig(r, key, jsonVal)
-			} else if isbool, _, _ := req.Option("bool").Bool(); isbool {
-				output, err = setConfig(r, key, value == "true")
-			} else {
-				output, err = setConfig(r, key, value)
+			var jsonVal interface{}
+			if err := json.Unmarshal([]byte(value), &jsonVal); err != nil {
+				err = fmt.Errorf("failed to unmarshal json. %s", err)
+				res.SetError(err, cmds.ErrNormal)
+				return
```

Comment:
```ipfs config Addresses.API /ip4/127.0.0.1/tcp/5001``` still works in this case (golang/pkg/encoding/json).

So I will have to check which implementation (RFC 4627 vs ECMA-262 5 vs ECMA-262 5.1) is among the majority today.